presto-main/src/main/java/io/prestosql/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -486,6 +495,33 @@ protected Node visitShowCreate(ShowCreate node, Void context)
                 return singleValueQuery("Create Table", formatSql(createTable).trim());
             }
 
+            if (node.getType() == SCHEMA) {
+                CatalogSchemaName schemaName = createCatalogSchemaName(session, node, Optional.of(node.getName()));
+                Optional<ConnectorSchemaDefinition> schema = metadata.getSchema(session, schemaName);
+
+                // TODO: at present everyone can see schema metadata
```

Comment:
This will need discussion. If only an owner or admin can do this, it's kinda pointless. On the other hand information can be leaked this way: Where the schema is located or that the assigned USER or ROLE exists.

Maybe limiting this to Owner/Admin is fine.